### ISSUE TITLE
Fonts sitting below baseline in Webkit browsers

### ISSUE THREAD

**Issue by andyra** [sapegin/grunt-webfont#233]
I have a group of Font Awesome SVGs I've converted into a webfont and the baseline is off in webkit browsers.

![font-icon-rendering](https://cloud.githubusercontent.com/assets/784440/6586526/3059b4ee-c748-11e4-901c-8f8b37ddc9be.png)

Here are things I tried to resolve this issue:

- Modify `fontHeight` and `descent` to various values (no effect)
- Modified SVGs in Illustrator to make sure the artboards were cropped correctly
- Minified the SVGs before compiling to webfont
- Played with every CSS value known to man on both the `.icon` and `.icon:before` elements.

I know it's not a browser-wide problem because other icon fonts render as expected (the icon font used on the Font Awesome website, for instance, renders the baselines correctly).

Baselines render correctly in Firefox, but the icons sit below the lines in Chrome and Safari. Any ideas what's up?

**Comment by sapegin** [comment]
Which engine do you use?

**Comment by andyra** [comment]
fontforge

**Comment by pensierinmusica** [comment]
I had a similar issue once... the following process might help you to fix the problem. Try to remove all svg files from your icons folder, then add only one file to your webfont and check if the problem persists.

Make sure that this single icon is a solid, one-color shape (usually black for convenience).

If the generated font with only one icon is fine, then the problem might be related to difference across icons. Checking the svg files in Adobe Illustrator or other vector graphics software, all canvases should have the same size (for example I use a 300 px square), and all icons centered in the canvas (both horizontally and vertically).

To keep rendering size consistent across icons I also suggest to have the largest side of the icon (i.e. height or width) equal to a constant amount, sligthly smaller than the canvas size (in my example 280 px).

Hope this helps!

**Comment by juarezpaf** [comment]
Having some issues also in Google Chrome... Mac rendering is above and Windows below.

![grunt-webfont-rendering](https://cloud.githubusercontent.com/assets/21221/6926016/e2bcb8da-d7b9-11e4-87f2-e23bc7d72e45.png)

**Comment by annahansen** [comment]
Have been facing the same problem today. Thank you @pensierinmusica for your ideas. When i compared the other SVGs to the broken one, i noticed that the broken SVG had width and height defined, the others did not have these attributes. So i removed the attributes and tried building the font again - now icons are at the right position again. I hope that's it. I'll let you know, if it wasn't.

**Comment by wayfarerboy** [comment]
I found the font to display differently between identical setups (identical browser and OS), but fixed it after following the ideas from @pensierinmusica then preparing the files using an approach I've taken when preparing svg files for embedding.

* Make sure all svgs have the same dimensions
* Ungroup everything in the document to simplify markup
* Run the files through `svgo` to simplify further
* Add a viewBox attribute to the svg tag of each file, eg. `viewBox="0 0 {width} {height}"` changing width and height to match document dimensions

This seems to be a pretty solid approach... so far!

**Comment by andyra** [comment]
Yep, I think these suggestions are the ticket. It's an SVG prep issue, not an issue with grunt-webfont. Thanks for the good work, people!

**Comment by gibbitz** [comment]
For those who are coming to this trying to correct baseline issues, I've noticed that in the current version (1.0.2), that below 100px and above 512px shifting occurs. I was using 36px square and read that this caused my fonts to not be generated (https://github.com/sapegin/grunt-webfont/issues/221) switched to 1024px and had problems (on unix systems the font was 1/2 height causing the baseline to shift up almost 50%). 

Tl;dr, the fix was to make the glyphs 512px square. It would be nice to have some best practices in the documentation so users wouldn't have to stumble around to figure out what the right prep for the SVGs are. Read more here (https://github.com/sapegin/grunt-webfont/issues/232 and here https://github.com/sapegin/grunt-webfont/issues/219)

**Comment by sapegin** [comment]
@gibbitz Yes, it’d nice. Pull requests are appreciated.
